pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1800,6 +1801,22 @@ func doActivateTail(ctx *domainContext, status *types.DomainStatus,
 
 	status.State = types.RUNNING
 	domainID, state, err := hyper.Task(status).Info(status.DomainName)
+	// If status is unknown, get info again for some retries and if it will not be activated return error.
+	for retry := 0; retry <= unknownStateRetries; retry++ {
+		if err == nil && state == types.UNKNOWN {
```

Comment:
```suggestion
		ps.StillRunning(agentName, warningTime, errorTime)
		if err == nil && state == types.UNKNOWN {
```
to make the watchdog happy

Reply:
@christoph-zededa @OhmSpectator IMO we don’t need to use StillRunning and create watchdog file every 2 seconds because warnTime (40s) and errTime (3min) are much greater than the total retry timeout (20s). Instead, we can call StillRunning once before the loop.

Your thoughts?

Reply:
Better safe than sorry, but fine for me.

So we assume that `hyper.Task(status).Info(status.DomainName)` is quite fast and does not take more than 2 seconds.

Reply:
hmm, it issues a containerD command to get the status, i haven't seen to hang for more than a second. However you never know, as you said, better safe than sorry!